### PR TITLE
[chore] remove calls to validate the configuration, now done in the config builder

### PR DESCRIPTION
**Pull request by atoulme**
**Description:**
As exposed in this [comment](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/16839#discussion_r1067636248), there is no longer a need to explicitly call `Validate()` on configuration when creating a component.
This PR removes occurrences of such calls.

### CODE REVIEW

**Comment by dmitryax** [comment]
Please update failing tests as well

**Comment by atoulme** [comment]
I'm on it ! Sorry, will keep iterating on it and get more PRs to help clean this up.

File: receiver/jmxreceiver/receiver.go [hunk 1]
```diff
@@ -119,6 +119,9 @@ func (jmx *jmxMetricReceiver) Start(ctx context.Context, host component.Host) er
 }
 
 func (jmx *jmxMetricReceiver) Shutdown(ctx context.Context) error {
+	if jmx.subprocess == nil {
+		return nil
+	}
```
Comment (dmitryax) [inline receiver/jmxreceiver/receiver.go:124]: Sounds like an unrelated chance? If so, can you please move it to a separate PR?

**Review by dmitryax — COMMENTED** [review]
Just couple of nits

File: receiver/jmxreceiver/receiver.go (hunk 1 above)
Comment (atoulme) [reply receiver/jmxreceiver/receiver.go:124]: Will do.

File: exporter/sumologicexporter/exporter_test.go [hunk 2]
```diff
@@ -54,20 +54,6 @@ func TestInitExporter(t *testing.T) {
 	assert.NoError(t, err)
 }
 
-func TestInitExporterInvalidConfig(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (atoulme) [reply exporter/sumologicexporter/exporter_test.go:57]: It's tested separately here: https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/e007c7239cd5fabd4efafa2f6c01d02115b397cb/exporter/sumologicexporter/config_test.go#L56

File: receiver/jmxreceiver/receiver.go (hunk 1 above)
Comment (atoulme) [reply receiver/jmxreceiver/receiver.go:124]: It's not unrelated actually. The receivers_test.go started to fail when I moved Validate out of the New function, because the receiver would be created but fail on shutdown.

I have peeled the change off to https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/17554 anyway.

**Review by dmitryax — APPROVED** [review]

**Comment by dmitryax** [comment]
I'll keep it open for some time to see if there is feedback from the code owners

**Review by djaglowski — APPROVED** [review]
